app/assets/javascripts/chat-lite/src/widgets/header-icons/chat-lite.gjs

```diff
@@ -0,0 +1,18 @@
+import { icon } from "discourse-plugin/helper";
+import { service } from "discourse-plugin/service";
+import Widget from "discourse-plugin/widget";
+import ChatService from "../../services/chat";
+
+export default class ChatLiteHeaderIcon extends Widget {
+  @service(ChatService) chat;
+
+  get isActive() {
+    return this.chat.userCanChat;
+  }
+
+  <template>
+    <li class="header-dropdown-toggle chat-header-icon">
```

Comment:
This part probably can't be easily moved into the template, which is unfortunate, but the "pre-instantiation" trick should allow for this being a property in the class – maybe even a reactive getter so that the widget framework can automatically update the wrapper without the function we currently pass in.